make/Makefile

```
@@ -0,0 +1,96 @@
+# Copyright 2021 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SOURCES=$(shell find . -type f -name "*.go")
+
+GOFLAGS=-ldflags '-w -s'
+# GOBUILDPROCS is passed to GOMAXPROCS when running go build
+GOBUILDPROCS=2
```

How can I set this to a bigger value?

---

You can!

```console
$ make -f make/Makefile -j8 GOBUILDPROCS=4 staged-release
```

That said, I'll actually change to this so that environment variables will also affect this variable:

```suggestion
GOBUILDPROCS ?= 2
```

Which would enable:

```console
$ export GOBUILDPROCS=4
$ make -f make/Makefile -j8 staged-release
```

---

Ahh, I did not know that passing a variable (`make VAR=foo`) was actually overriding variable assignments of the form `VAR=bar` in the makefile itself!

By the way, what is the rationale for not letting Go use the available cores? Is it to avoid overloading when using `-j8`?

---

> By the way, what is the rationale for not letting Go use the available cores? Is it to avoid overloading when using -j8?

Exactly, yeah. With `-jN` we can be running `go build` N times in parallel, and GOMAXPROCS defaults to the number of threads available on the system so it's just too much. This is a pain point with make; ideally I think we'd dynamically set GOMAXPROCS lower if running parallel but make doesn't really expose that information at runtime unfortunately (the value of `N` is _obtainable_ but it's hacky as hell).

It might make sense to have the default for `GOBUILDPROCS` be empty, so that the default behaviour is to use all cores when building individual binaries for testing on a laptop. We'd then manually set `GOBUILDPROCS` lower when running `make -f make/Makefile -jN staged-release` in cloudbuild.